### PR TITLE
Issue #DP-844 feat: Save as a Draft

### PR DESCRIPTION
**Pull request by ahghatol**

### CODE REVIEW

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts [hunk 1]
```diff
@@ -33,9 +34,11 @@ export class CreateProgramComponent implements OnInit, AfterViewInit {
   public programId: string;
   public guidLinefileName: String;
   public isFormValueSet = false;
-  public editMode = false;
+  public editDraft = false;
```
Comment (twsvaishali) [inline src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:37]: @ahghatol editDraft is nothing but !editLive? Can we reduce one variable?

Comment (ahghatol) [reply src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:37]: I think we should not do that

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { collection } from './../list-contributor-textbooks/data';
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:1]: Why are we importing this?

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts [hunk 3]
```diff
@@ -704,71 +722,121 @@ export class CreateProgramComponent implements OnInit, AfterViewInit {
...
+      // tslint:disable-next-line:max-line-length
+      _.find(_.find(this.programConfig.components, { id: 'ng.sunbird.collection' }).config.filters.implicit, { code: 'framework' }).defaultValue = this.userFramework;
+    }
 
-      if (_.isEmpty(this.programData.config.board) &&
-      _.findIndex(this.frameworkCategories, function(item) { return item.code === 'board'; }) < 0) {
-        delete this.programData.config.board;
-      }
+    // tslint:disable-next-line: max-line-length
+    _.find(_.find(this.programConfig.components, { id: 'ng.sunbird.collection' }).config.filters.implicit, { code: 'board' }).defaultValue = this.userBoard;
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:765]: Should we set it to `value` or `defaultValue`? 

also, are we not setting the board to `this.programConfig.board` like how you are setting `this.programConfig.framework` ?

Comment (venkateshwarans) [reply src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:765]: @twsvaishali or @pallakartheekreddy  correct me If I am wrong.

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts (hunk 2 above)
Comment (ahghatol) [reply src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:1]: Not needed, I'll remove

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts (hunk 3 above)
Comment (ahghatol) [reply src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:765]: @venkateshwarans it has defaultValue proptery. there is value property in `this.programConfig.components`,

Comment (ahghatol) [reply src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:765]: Screenshot from 2020-07-30 21-17-12

Comment (ahghatol) [reply src/app/client/src/app/modules/program/components/create-program/create-program.component.ts:765]: @venkateshwarans We are setting board too, please refer line number 822
`
this.programConfig['board'] = config.board;`

File: src/app/client/src/app/modules/program/components/create-program/create-program.component.ts (hunk 1 above)
Comment (ahghatol) [reply src/app/client/src/app/modules/program/components/create-program/create-program.component.ts]: Fixed

**Pull request merged**